common/djangoapps/util/milestones_helpers.py

```diff
@@ -143,6 +145,36 @@ def fulfill_course_milestone(course_key, user):
             add_user_milestone({'id': user.id}, milestone)
 
 
+def calculate_entrance_exam_score(user, course_descriptor, exam_modules):
+    """
+    Calculates the score (percent) of the entrance exam using the provided modules
+    """
+    exam_module_ids = [exam_module.location for exam_module in exam_modules]
+    student_modules = StudentModule.objects.filter(
+        student=user,
+        course_id=course_descriptor.id,
+        module_state_key__in=exam_module_ids,
+    )
+    print exam_module_ids
+    exam_pct = 0
+    if student_modules:
+        module_pcts = []
+        ignore_categories = ['course', 'chapter', 'sequential', 'vertical']
```

Comment:
<sigh> we use this pattern all over the codebase - Marty had something similar, I wonder if we should make a helper utility - if it doesn't exist - something like

module.category.is_container()

Or someone consolidate the logic regarding identifying what categories are considered containers.

Anyhow, this isn't pertinent to this PR, just grumbling...